Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java

```diff
@@ -17,79 +21,205 @@
+            }
+
+            Node objectNode = link.getObject();
+            if (!(objectNode instanceof PredicateNode)) {
+                link.setObject(updatedNodes.get(objectNode.getConceptId()));
+            }
+        }
+
+        // return new network with updated nodes and links
+        return new TransformedNetwork(updatedNodes, links);
+    }
+
+    @Override
+    public ITransformedNetwork getSearchTransformedNetwork(String projectId, String conceptId)
+        throws QuadrigaStorageException {
+        // get the transformed network and search for the concept id
+        List<INetwork> networkList = getNetworkList(projectId);
+
+        if (networkList == null) {
+            return null;
+        }
+
+        ITransformedNetwork transformedNetwork = getTransformedNetworkOfProject(projectId);
+        // separate each statement id and include all the list of nodes in the statement id
+        Map<String, Set<Node>> statementIdMap = getStatementIdNodeMap(transformedNetwork.getNodes());
+
+        // Filter the nodes with the concept id
+        List<Node> searchNodes = new ArrayList<Node>();
+        for (Node node: transformedNetwork.getNodes().values()) {
+            if (conceptId.equals(node.getConceptId())) {
+                searchNodes.add(node);
+            }
+        }
+
+        // include only those statements which have the search node
+        Iterator<Map.Entry<String, Set<Node>>> it = statementIdMap.entrySet().iterator();
+        Set<Node> searchedNodeSet = new HashSet<Node>();
+        while (it.hasNext()) {
+            boolean isNodeExists = false;
+            Set<Node> nodeSet = it.next().getValue();
+            // search if the node exists
+            for (Node node: searchNodes) {
+                if (nodeSet.contains(node)) {
+                    // include this statement
+                    isNodeExists = true;
+                }
+            }
+            // remove the statement if it is not present
+            if (!isNodeExists) {
+                it.remove();
+            } else {
+                searchedNodeSet.addAll(nodeSet);
+            }
+        }
+
+        // now include the statements
+        Map<String, Node> finalNodes = new HashMap<String, Node>();
+        int index = 0;
+        for (Node n: searchedNodeSet) {
+            finalNodes.put(String.valueOf(++index), n);
+        }
+        // final links
+        List<Link> finalLinks = transformedNetwork.getLinks();
+        Iterator<Link> linkIterator = finalLinks.iterator();
+        while (linkIterator.hasNext()) {
+            Link link = linkIterator.next();
+            // if the subject or object does not exist in any links
+            // remove that link
+            if (!searchedNodeSet.contains(link.getSubject()) ||
+                    !searchedNodeSet.contains(link.getObject())) {
+                linkIterator.remove();
+            }
+        }
```

Comment:
If you add a statement id field to the Link class and modify the eventparser accordingly then you can do the same thing (as described above, filtering by statement id) with links.